traits/ctraits.c

```diff
@@ -3064,6 +3064,22 @@ _trait_set_default_value(trait_object *trait, PyObject *args)
         return NULL;
     }
 
+    /* Validate the value */
+    switch (value_type) {
+        /* We only do sufficient validation to avoid segfaults when
+           unwrapping the value in `default_value_for`. */
+        case CALLABLE_AND_ARGS_DEFAULT_VALUE:
+            if (!PyTuple_Check(value) || PyTuple_GET_SIZE(value) != 3) {
+                PyErr_SetString(
+                    PyExc_ValueError,
+                    "default value for type DefaultValue.callable_and_args "
+                    "must be a tuple of length 3"
```

Comment:
It would be friendlier to remind the user of the expected format `(callable, args, kwds)` instead of just specifying the length.

Reply:
Done in https://github.com/enthought/traits/pull/1631/commits/ffeb4e23e915a8be47dbe7ff73ad69ef4dc12be1